### ISSUE TITLE
Feature : Add border to dock 

### ISSUE THREAD

**Issue by felipedobrinin** [micheleg/dash-to-dock#1883]
Be able to add border with pixel count, type and color. At the moment the only way of doing this is through the stylesheet.css.
This makes it hard for non technical people to add a simple border.

**Comment by stpnwf** [comment]
I second that. I would like to add 2 1px borders to the dock to get the [Colloid](https://github.com/vinceliuice/Colloid-gtk-theme) outline appearance.

In the meantime, @felipedobrinin how would one add borders through the stylesheet.css?

**Comment by vanvugt** [comment]
See also #1899

**Comment by bedxnta** [comment]
i 3rd this. would be a goated option ngl fr

**Comment by xxxtrashcion** [comment]
4th this 2 years later.

**Comment by Meemonomon** [comment]
I 5th this

**Comment by bedxnta** [comment]
i no longer use linux sorry guys i got a new laptop and i want to play my dream games on it... so sorry for betrayal i feel like shit rn but it is what it is.

**Comment by MeechaPooch** [comment]
@bedxnta you okay?

**Comment by bedxnta** [comment]
> [@bedxnta](https://github.com/bedxnta) you okay?

i don't think i am. i tried to kill myself last week and the girl i liked is not talking to me these days because she has to focus on her exams. i talked to her brother and he said not to worry and her blocking me is temporary, and that he'd ask her to unblock me once their exams are over (both of them have exams).

**Comment by MeechaPooch** [comment]
Who do you have to talk to?

**Comment by bedxnta** [comment]
i don't know really, can't discuss any of this with my parents because they're very conservative. maybe one or two of my friends but I don't wanna tell them directly. probably just skim by everything and say that "hey im sad" or something

**Comment by MeechaPooch** [comment]
Therapy?

**Comment by MeechaPooch** [comment]
Your profile says you are a student. Do you go to a school with a health center?

**Comment by bedxnta** [comment]
the place where i live, therapy is very far fetched and that would require money. also, since I don't have a degree i can't get a job.

**Comment by bedxnta** [comment]
and no my college has no such place

**Comment by MeechaPooch** [comment]
Do you know anyone who has struggled with suicide and depression but got through and are doing better? I have a friend who is so fucking cool, but she told me she once almost killed herself. I am hanging on by a thread these days too man. Hanging out with her helps me a little.

**Comment by MeechaPooch** [comment]
If any of those friends seem like they could relate, opening up to them could help you. One conversation can open up so much talking about it. But I think it definitely helps to talk with someone who has struggled too.

**Comment by bedxnta** [comment]
nope, my friends are healthy as they should be. the girl i was talking about earlier is a bit like me but not suicidal, but she's way too stressed about her exams so ig that's why she did that. i will trust her brother and hope that she unblocks me after her exams.

**Comment by MeechaPooch** [comment]
I am currently trying to get medication. Have you thought about medication for depression/adhd/ocd. Do you have any symptoms of these disorders or any diagnoses?

**Comment by bedxnta** [comment]
i might have them but medication is hard to come by.

**Comment by MeechaPooch** [comment]
What makes you feel better for however ammount of time?

**Comment by bedxnta** [comment]
nothing really, im about to go to sleep.

**Comment by MeechaPooch** [comment]
I could talk to you tomorrow if you'd like. Later in the day. Its 9:30pm for me now.

**Comment by bedxnta** [comment]
i will let you know.

**Comment by MeechaPooch** [comment]
okay

**Comment by MeechaPooch** [comment]
sleep good

**Comment by bedxnta** [comment]
hey man, just letting you know that the girl i talked about, im talking to her again. she was just stressed about exams so she went haywire.

im doing fine too.

**Comment by MeechaPooch** [comment]
Thats very good to hear :) Good luck dude :)

**Comment by ghost** [comment]
I use the following method to apply a custom border, which survives applying custom colors & opacity (a problem mentioned in workaround issue #2332).

1. Edit stylesheet.css & search for one of the following 4 sections (depending upon your preferred dock position):
  #dashtodockContainer.bottom #dash #dashtodockDashContainer
  #dashtodockContainer.top #dash #dashtodockDashContainer
  #dashtodockContainer.left #dash #dashtodockDashContainer
  #dashtodockContainer.right #dash #dashtodockDashContainer
2. Add the following 3 elements (customize for yourself, use rgba if preferred):
    border: 1px;
    border-color: white;
    border-radius: 24px;

Example: I place the dock on the bottom, so I edit the stylesheet.css section starting at line 29 so that it looks like this:

  #dashtodockContainer.bottom #dash #dashtodockDashContainer {
    border: 2px;
    border-color: rgba(100, 100, 100, 1.0);
    border-radius: 24px;
    padding: 10px;
    padding-bottom: 0;
    padding-top: 0; }

I apply with a simple sed one-liner:
sed -i '/#dashtodockContainer.bottom #dash #dashtodockDashContainer/a\\ \ \ \ border: 2px;\n\ \ \ \ border-color: rgba(100, 100, 100, 1.0);\n\ \ \ \ border-radius: 24px;' stylesheet.css
